gpt_explainer/gpt_slide_expander.py

```diff
@@ -0,0 +1,272 @@
+import asyncio
+import json
+import os
+import threading
+import time
+from typing import Dict, List
+from pptx_parser.pptx_parser import PPTXParser
+import openai
+
+
+class GptSlideExpander:
+    """
+    A class that utilizes OpenAI's GPT models to generate expanded explanations for presentation slides.
+
+    Attributes:
+    expanded_slide_explanations (Dict[int, str]): A dictionary that maps slide indices to their generated explanations.
+    The keys represent the slide index, and the values are the corresponding expanded explanations for each slide.
+    This dictionary is populated when `generate_explanation_for_presentation` is called.
+    """
+
+    def __init__(self):
+        GptSlideExpander.configure_api_key()
+        self.expanded_slide_explanations = {}
+        self.processed_files_lock = threading.Lock()
+        self.uploads_folder = ""
+        self.output_folder = ""
+
+    def set_output_folder(self, output_path: str) -> None:
+        self.output_folder = output_path
+
+    def set_uploads_folder(self, uploads_path: str) -> None:
+        self.uploads_folder = uploads_path
+
+    @staticmethod
+    def configure_api_key():
+        """Configure OpenAI API key."""
+        openai.api_key = os.getenv("OPENAI_API_KEY")
+
+    @staticmethod
+    def generate_presentation_main_topic(presentation_text: str, max_retry: int) -> str:
```

Comment:
Separate code to small functions.  Read more: https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be